gk/main.c

```diff
@@ -582,9 +723,9 @@ gk_proc(void *arg)
 }
 
 struct gk_config *
-alloc_gk_conf(void)
+get_gk_conf(void)
```

Comment:
That is not a sound justification for making that change.

What you are saying is that GGU needs a reference to `struct gk_config`. So why didn't you change `run_ggu()` to require that as another parameter. Not only would it solve the problem, it would also make clear that you can only have the GGU if there is at least a GK block.